services/issue/issue.go

```diff
@@ -55,6 +55,41 @@ func ChangeTitle(issue *models.Issue, doer *user_model.User, title string) (err
 	return nil
 }
 
+// ChangeConfidential changes the confidential of this issue, as the given user.
+func ChangeConfidential(issue *models.Issue, doer *user_model.User, isConfidential bool) (err error) {
```

Comment:
(duplicated comment) I think we should unify the concept, use `private` or `confidential` only, instead of mixing them.